<robot name="first_robot">
	
	<link name="base" >
		<visual>
			<geometry>
				<cylinder radius="0.1" length="0.06"/>
			</geometry>
			<material name="black">
				<color rgba="0 0 0 0.8"/>
			</material>
			<origin xyz="0 0 0.03" />
		</visual>
	</link>

	<link name="base_2" >
		<visual>
			<geometry>
				<cylinder radius="0.03" length="0.6"/>
			</geometry>
			<material name="red">
				<color rgba="255 0 0 1"/>
			</material>
			<origin xyz="0 0 0.2" />
		</visual>
	</link>

	<link name="link1" >
		<visual>
			<geometry>
				<box size="1 0.1 0.01" />
			</geometry>
			<material name="olive">
				<color rgba="128 128 0 0.8"/>
			</material>
			<origin xyz="0.5 0 0" />
		</visual>
	</link>

	<link name="link1_1" >
		<visual>
			<geometry>
				<cylinder radius="0.05" length="0.01" />
			</geometry>
			<material name="olive">
				<color rgba="128 128 0 0.8"/>
			</material>
		</visual>
	</link>

	<link name="link1_2" >
		<visual>
			<geometry>
				<cylinder radius="0.05" length="0.01" />
			</geometry>
			<material name="olive">
				<color rgba="128 128 0 0.8"/>
			</material>
		</visual>
	</link>

	<link name="link2_1" >
		<visual>
			<geometry>
				<cylinder radius="0.05" length="0.01" />
			</geometry>
			<material name="blue">
				<color rgba="0 0 .8 1"/>
			</material>
		</visual>
	</link>

	<link name="link2" >
		<visual>
			<geometry>
				<box size="1 0.1 0.01" />
			</geometry>
			<material name="blue">
				<color rgba="0 0 .8 1"/>
			</material>
			<origin xyz="0.5 0 0" />
		</visual>
	</link>

	<link name="end" >
		<visual>
			<geometry>
				<sphere radius="0.08" />
			</geometry>
			<material name="red">
				<color rgba="255 0 0 1"/>
			</material>
			<origin xyz="0 0 0" />
		</visual>
	</link>

	
	<joint name="base_joint" type="fixed">
		<parent link="base" />
		<child link="base_2" />
		<origin xyz="0 0 0" rpy="0 0 0" />	
	</joint>

	<joint name="joint1" type="continuous">
		<parent link="base_2" />
		<child link="link1_1" />
		<origin xyz="0 0 0.065" rpy="0 0 0" />	
		<axis xyz="0 0 1" />	
	</joint>

	<joint name="joint1_1" type="fixed">
		<parent link="link1_1" />
		<child link="link1" />
		<origin xyz="0 0 0" rpy="0 0 0" />
	</joint>

	<joint name="joint1_2" type="fixed">
		<parent link="link1" />
		<child link="link1_2" />
		<origin xyz="1 0 0" rpy="0 0 0" />
	</joint>	

	<joint name="joint2" type="continuous">
		<parent link="link1_2" />
		<child link="link2_1" />
		<origin xyz="0 0 0.01" rpy="0 0 0" />
		<axis xyz="0 0 1" />
	</joint>

	<joint name="joint2_1" type="fixed">
		<parent link="link2_1" />
		<child link="link2" />
		<origin xyz="0 0 0" rpy="0 0 0" />
	</joint>

	<joint name="end_joint" type="fixed">
		<parent link="link2" />
		<child link="end" />
		<origin xyz="1 0 0" rpy="0 0 0" />
	</joint>
	
</robot>
